### PR TITLE
api: split connection string into 2 new fields

### PR DESCRIPTION
**Pull request by iamniting**
remove connectionString and create 2 new fields:
- OnboardingTicket
- ExternalServerAddress

Signed-off-by: Nitin Goyal <nigoyal@redhat.com>

### CODE REVIEW

File: api/v1/storagecluster_types.go [hunk 1]
```diff
@@ -162,9 +162,11 @@ type ExternalStorageClusterSpec struct {
 	// StorageProviderKind Identify the type of storage provider cluster this consumer cluster is going to connect to.
 	StorageProviderKind ExternalStorageKind `json:"storageProviderKind,omitempty"`
 
-	// ConnectionString An encoded string holding connection and identity information
-	// that is needed in order to establish connection with the storage providing cluster.
-	ConnectionString string `json:"connectionString,omitempty"`
```
Comment (sp98) [inline api/v1/storagecluster_types.go:167]: The code that removes usage of  `ConnectionString` should  also have been in the same commit. Right now its removed in the third commit.

**Review by nb-ohad — CHANGES_REQUESTED** [review]

File: api/v1/storagecluster_types.go [hunk 2]
```diff
@@ -162,9 +162,11 @@ type ExternalStorageClusterSpec struct {
...
+	// OnboardingTicket holds an identity information required for consumer to onboard.
+	OnboardingTicket string `json:"onboardingTicket,omitempty"`
+
+	// ExternalServerAddress holds info to establish connection with the storage providing cluster.
+	ExternalServerAddress string `json:"externalServerAddress,omitempty"`
```
[the remark this answers is not in the document]
Comment (iamniting) [reply api/v1/storagecluster_types.go]: changed the name

File: api/v1/storagecluster_types.go (hunk 1 above)
Comment (iamniting) [reply api/v1/storagecluster_types.go:167]: Actually, we like to keep API and controller changes in the different commits.

**Review by nb-ohad — APPROVED** [review]

File: api/v1/storagecluster_types.go (hunk 1 above)
Comment (sp98) [reply api/v1/storagecluster_types.go:167]: This is a different scenario.  An API field is removed that is being used in the controller.  This commit on its own won't build. IMO, `ConnectionString` and all its references should have been removed in a single PR. 
Approving the PR.

**Review by sp98 — APPROVED** [review]

**Pull request merged**